dfastmi/batch/AnalyserAndReporterWaqua.py

```diff
@@ -26,29 +26,25 @@
 INFORMATION
 This file is part of D-FAST Morphological Impact: https://github.com/Deltares/D-FAST_Morphological_Impact
 """
-
 from typing import Optional, List, Union, Tuple, TextIO
-from dfastmi.kernel.core import Vector, BoolVector
-from dfastmi.io.ApplicationSettingsHelper import ApplicationSettingsHelper
-from dfastmi.io.DataTextFileOperations import DataTextFileOperations
-
 import os
 import numpy
+
 import dfastmi.kernel.core
 import dfastmi.plotting
+from dfastmi.kernel.core import Vector, BoolVector
+from dfastmi.io.ApplicationSettingsHelper import ApplicationSettingsHelper
+from dfastmi.io.DataTextFileOperations import DataTextFileOperations
 
 def analyse_and_report_waqua(
     display: bool,
     report: TextIO,
     reduced_output: bool,
```

Comment:
Kunnen deze niet bij elkaar? in de functie call oid?

Reply:
of een soort van logger functionaliteit

Reply:
```
# -*- coding: utf-8 -*-
"""
Copyright (C) 2024 Stichting Deltares.

This library is free software; you can redistribute it and/or
modify it under the terms of the GNU Lesser General Public
License as published by the Free Software Foundation version 2.1.

This library is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
Lesser General Public License for more details.

You should have received a copy of the GNU Lesser General Public
License along with this library; if not, see <http://www.gnu.org/licenses/>.

contact: delft3d.support@deltares.nl
Stichting Deltares
P.O. Box 177
2600 MH Delft, The Netherlands

All indications and logos of, and references to, "Delft3D" and "Deltares"
are registered trademarks of Stichting Deltares, and remain the property of
Stichting Deltares. All rights reserved.

INFORMATION
This file is part of D-FAST Morphological Impact: https://github.com/Deltares/D-FAST_Morphological_Impact
"""
"""
Module for ISimpleLogger interface and SimpleLogger implementation

Interfaces:
    ISimpleLogger

"""

from abc import ABC, abstractmethod
from typing import TextIO

from dfastmi.io.ApplicationSettingsHelper import ApplicationSettingsHelper

class ISimpleLogger(ABC):
    """Interface for logging"""  

    @abstractmethod
    def Log(self, message : str):
        """Log method to log the message"""

class SimpleLogger(ISimpleLogger):
    """Log a message"""
    _report: TextIO
    _write_to_std_out : bool
    reduced_output : bool

    def __init__(self, report : TextIO, write_to_std_out : bool = False, reduced_output : bool = False):
        """
        Create Branch based on name. 
        Initialize the reaches

        Args:
            report : TextIO
                Text stream for log file.
            write_to_std_out : bool
                Flag indicating text output to stdout.
            reduced_output : bool
                Flag to indicate whether WAQUA output should be reduced to the area of
        interest only.
        """

        self._report = report
        self._write_to_std_out = write_to_std_out
        self._reduced_output = reduced_output
    
    def Log(self, message: str):
        if self._write_to_std_out:
            ApplicationSettingsHelper.log_text(str)
```
Or maybe look into https://github.com/Deltares/D-EcoImpact/tree/main/decoimpact/crosscutting

Reply:
Please ignore / decide for you self if it is worth the trouble, energy/ time for this issue but keep in mind for when working on dflow analyser.

Reply:
maybe we should discuss this in more detail either in a call or face to face

Reply:
moved all logging calls to a small separate private class